server/api/user/scroll/index.ts

```diff
@@ -103,16 +104,23 @@ export default defineEventHandler(async (event) => {
     details: {
       clicksToday,
     },
-    dragons: alive.map<ScrollView>((id) => {
-      const hatcheryDragon = usersDragonsInHatchery.find(
-        (row) => row.id === id
-      );
+    dragons: alive
+      .map<ScrollView>((id) => {
+        const hatcheryDragon = usersDragonsInHatchery.find(
+          (row) => row.id === id
+        );
+
+        return {
+          ...scrollResponse.dragons[id],
+          in_garden: !!(hatcheryDragon?.in_garden ?? false),
+          in_seed_tray: !!(hatcheryDragon?.in_seed_tray ?? false),
+        };
+      })
+      .map((dragon) => {
+        dragon.incubated = isIncubated(dragon);
+        dragon.stunned = isStunned(dragon);
 
-      return {
-        ...scrollResponse.dragons[id],
-        in_garden: !!(hatcheryDragon?.in_garden ?? false),
-        in_seed_tray: !!(hatcheryDragon?.in_seed_tray ?? false),
-      };
-    }),
+        return dragon;
+      }),
```

Comment:
I mean, this obviously works and isn't necessarily wrong as such, but you're mutating the value and returning it when map is really about returning a new value, no?

Reply:
🤷

Reply:
hold my beer